### ISSUE TITLE
Cylinder of same radius having different radius during visualization

### ISSUE THREAD

**Issue by lokitkhemka** [zalo/CascadeStudio#148]
When, I was running the following code: 

```
function DrawCylinder(Center, Radius, Length, Angle,  Direction)
{
    let shape = Cylinder(Radius, Length, false);
    shape = Rotate(Direction, Angle,shape, false);
    shape = Translate(Center, shape, false);
    return shape
}
let scale = 2.0;
let Radius = Slider("Radius", 0.65 , 0.5 , 1.0);

let RefPipe = DrawCylinder([0,0,0], Radius *scale, 4*scale,0, [1,0,0]);
let Pipe1 = DrawCylinder([8*scale,75*scale,8*scale], Radius *scale, 4*scale, -90, [1,0,0]);
let Pipe2 = DrawCylinder([8*scale,79*scale,8*scale], Radius *scale, 12*scale, 0, [1,0,0]);
let Pipe3 = DrawCylinder([8*scale,79*scale,20*scale], Radius *scale, 37*scale, 90, [1,0,0]);
```

Pipe2 has a vastly different radius than rest of the cylinders. This is on google chrome.

**Comment by raydeleu** [comment]
There are strange things happening. I used the following steps: 
* started CascadeStudio as a web application
* copied the code above into the editor window

I then got the following result.
![image](https://github.com/zalo/CascadeStudio/assets/38007983/40f03eac-b619-4fb0-9d19-0364d4628ba0)

I then removed the checkmark at the ```cache```, immediately it changes to the following result: 

![image](https://github.com/zalo/CascadeStudio/assets/38007983/82d0be43-35f4-4962-8505-b46b8e163a10)

Setting the cache again to enabled changes the result again: 

![image](https://github.com/zalo/CascadeStudio/assets/38007983/5f125464-fcd0-40ad-8c40-839347eb8b77)

Another way to see the change is to keep the cache checkmark but to put comment marks around some of the shapes and then remove these comment marks again. I suppose that this also flushes the cache for these shapes. 
My assumption is that your function makes it difficult for the caching mechanism to recognize when a shape has to be recalculated. Leaving the cache off seems to solve all issues.
